src/common.rkt

```diff
@@ -170,16 +170,13 @@
    [(and (> r 0) sign) (format "+~a~a" (/ (round (* r 10)) 10) unit)]
    [else (format "~a~a" (/ (round (* r 10)) 10) unit)]))
 
-(define (format-accuracy r repr #:sign [sign #f] #:unit [unit #f])
-  (cond 
-    [(not r) ""]
-    [else
-      (define unit- (if unit unit ""))
-      (define percent (~r (- 100 (* (/ r (representation-total-bits repr)) 100)) #:precision '(= 1)))
-
-      (cond
-      [(and (> r 0) sign) (format "+~a~a" percent unit-)]
-      [else (format "~a~a" percent unit-)])]))
+(define (format-accuracy numerator denominator #:sign [sign #f] #:unit [unit ""])
+  (if numerator
+      (let ([percent (~r (- 100 (* (/ numerator denominator) 100)) #:precision '(= 1))])
+        (if (and (> numerator 0) sign)
+            (format "+~a~a" percent unit)
+            (format "~a~a" percent unit)))
+      ""))
```

Comment:
I think it's possible for the denominator to be zero (consider a single benchmark with no error, like `(FPCore () 1)`. We don't want to crash in that case.

Reply:
So I ran Herbie with that input and it worked. I think it worked because denominator actually gets passed in as 64. I'm calculating accuracy by summing `table-row-start`, `table-row-result`, and `table-row-precision` for each test, and the resulting string is basically `(format "~a -> ~a" (/ (sum starts) (sum representation-total-bits)) (/ (sum results) (sum representation-total-bits))` but with more specific formatting from `format-accuracy`. It seems to give the right results, but is that right?

Regardless, I added a check for the denominator being `positive?`, and if it fails it just returns an empty string like it already did if the numerator was `null?`. Does that seem reasonable?

Reply:
![Uploading Screen Shot 2023-05-30 at 4.06.24 PM.png…]()

Reply:
Oh, ok. If the denominator is representation-total-bits then it shouldn't ever be 0. But the check is fine too, empty string seems ok.